app/views/contacts/index.html.erb

```diff
@@ -1,45 +1,89 @@
-      </tr>
+<h2 class="h2 offset-1">Your contacts 
+  <span class="badge">
+    <%= link_to new_contact_path, class: "btn btn-link btn-sm text-muted" do %>
+    <i class="bi bi-person-plus-fill" style="font-size: 2rem"></i>
     <% end %>
-  </tbody>
-</table>
+  </span>  
+</h2>
+
+<div class="container">
+  <%= search_form_for @q do |f| %>
+    <h4>Search for a contact:</h4>
+    
+      <div class="row justify-content-left offset-1">
+        <div class="col-md-3">
+          <%= f.label :name_cont, "Name containing" %>
+          <%= f.text_field :name_cont, :placeholder => "Enter a few characters" %>
+        </div>
+
+        <div class="col-md-3">
+          <%= f.label :last_name_cont, "Last name containing" %>
+          <%= f.text_field :last_name_cont, :placeholder => "Enter a few characters" %>
+        </div>
+
+        <div class="col-md-3">
+          <%= f.label :organization_cont, "Organization containing" %>
+          <%= f.text_field :organization_cont, :placeholder => "Enter a few characters" %>
+        </div>
+
+        <div class="col-md-3">
+          <%= f.label :phone_number_cont, "Phone number containing" %>
+          <%= f.text_field :phone_number_cont, :placeholder => "Enter a few characters" %>
+        </div>
+      </div>
+    
+      <div class="row mt-3 justify-content-left offset-1">
+        <%= f.submit :class => "btn btn-primary btn-block" %>
+      </div>
+      
+      <div class="row mt-1 mb-3 justify-content-left offset-1">
+        <a href="/contacts">Clear filters</a>
+      </div>  
+
+  <% end %>
+</div>
+
+<div class= "container-flex">
+  <table class ="table">
+    <thead>
+      <tr class="row justify-content-center">
+        <th class="col-md-1">Name</th>
+        <th class="col-md-1">Last name</th>
+        <th class="col-md-2">Phone number</th>
+        <th class="col-md-1">Channel</th>
+        <th class="col-md-2">Email</th>
+        <th class="col-md-1">Organization</th>
+        <th class="col-md-1">Social Networks</th>
+        <th class="col-md-1"></th>
+      </tr>
+    </thead>
+
+    <tbody>
+      <% @contacts.each do |contact| %>
+        <tr class="row justify-content-center">
+          <td class="col-md-1"><%= contact.name %></td>
+          <td class="col-md-1"><%= contact.last_name %></td>
+          <td class="col-md-2"><%= contact.phone_number %></td>
+          <td class="col-md-1"><%= contact.communication_channel %></td>
+          <td class="col-md-2"><%= contact.email %></td>
+          <td class="col-md-1"><%= contact.organization %></td>
+          <td class="col-md-1">Show social Networks</td>
+          <td class="col-md-1">
+
+            <%= link_to edit_contact_path(contact), class: "btn btn-link btn-sm text-muted" do %>
+              <i class="bi bi-pencil-fill"></i>
+            <% end %>
+
+            <%= link_to contact, method: :delete, class: "btn btn-link btn-sm text-muted" do %>
+              <i class="bi bi-trash-fill"></i>
+            <% end %>
```

Comment:
I am not really sure if this is the way to make a table with the proper spacing

Reply:
What kind of spacing are you trying to do?  Do you mean the spacing/indenting in this file or the CSS spacing when you visit the page? Or something else?

Reply:
I mean CSS spacing, I don't know if the table is adapting properly to its contents as the width of each column is kind of fixed.